text/0748-glimmer-component-signature.md

```diff
@@ -129,7 +124,32 @@ A signature with no `Args` indicates that its component does not accept any argu
 
 ### `BlockParams`
 
-The `BlockParams` member dictates what blocks a component accepts and specifies what parameters, if any, it provides to those blocks.
+The `BlockParams` member dictates what blocks a component accepts and specifies what parameters, if any, it provides to those blocks. When `BlockParams` is a simple tuple type, that indicates that the component only yields to the default block:
+
+```ts
+BlockParams: [name: string];
+```
+
+```hbs
+{{yield "Tomster"}}
+```
+
+However, `BlockParams` may also be an object type indicating what parameters a component's named blocks provide:
+
+```ts
+BlockParams: {
+  header: [];
+  body: [item: T; index: number]
+}
+```
+
+```hbs
+{{yield to="header"}}
+
+{{#each items as |item index|}}
+  {{yield item index to="body"}}
+{{/each}}
+```
```

Comment:
We should probably include how we handle `default` in the extended case as well. (Note that a "suggestion" doesn't work here b/c of the nested code blocks, so I'm just going to "quote" the suggested text instead.)

> However, `BlockParams` may also be an object type indicating what parameters a component's named blocks provide. In that case, the `default` block may be specified by name:
> 
> ```ts
> BlockParams: {
>   header: [];
>   default: [];
>   listItem: [item: T; index: number]
> }
> ```
> 
> ```hbs
> <h1>{{yield to="header"}}</h1>
> 
> <div class='body'>
>   {{yield}}
> </div>
> 
> <ul class='items'>
>   {{#each items as |item index|}}
>     <li class='item'>{{yield item index to="listItem"}}</li>
>   {{/each}}
> </ul>
> ```